packages/nimble-components/src/table/models/tests/sort-operations.spec.ts

```diff
@@ -0,0 +1,93 @@
+import type { Row as TanStackRow, SortingFn } from '@tanstack/table-core';
+import { TableColumnSortOperation } from '../../../table-column/base/types';
+import { getTanStackSortingFunction } from '../sort-operations';
+import type { TableFieldValue } from '../../types';
+import { parameterizeNamedList } from '../../../utilities/tests/parameterized';
+
+describe('Sort Operations', () => {
+    function createFakeRow(
+        value: TableFieldValue
+    ): TanStackRow<TableFieldValue> {
+        const row = jasmine.createSpyObj<TanStackRow<TableFieldValue>>(
+            'TanStackRow',
+            ['getValue']
+        );
+        row.getValue.and.returnValue(value);
+        return row;
+    }
+
+    // The `expectedSortOrder` arrays in the test cases are ordered from smallest value
+    // to largest value. The tests call the sort operation on each index of each array
+    // with every index in the same array and verify that it returns the correct value.
+    const testCases = [
+        {
+            name: 'basic sorting with numbers',
+            sortOperation: TableColumnSortOperation.basic,
+            expectedSortOrder: [
+                undefined,
```

Comment:
I think we should also be testing `NaN` since it is a weird one as `NaN !== NaN` so that case should behave well

Reply:
I've added handling and testing for `NaN`. I am sorting it between `null` and `NEGATIVE_INFINITY`, so let me know if you think it would be more appropriately sorted somewhere else.

Reply:
Seems like a good spot! The most number like not a number.

Reply:
I guess if we are also being complete with the IEEE754 edge cases we should show how `-0` will be sorted (a slightly smaller zero :P)